Update the RWKV documentation with fixes to spelling and wording (#24168)
huggingface/transformers: docs/source/en/model_doc/rwkv.mdx

## Overview

The RWKV model was proposed in [this repository](https://github.com/BlinkDL/RWKV-LM)

> It suggests a tweak in the traditional Transformer attention to make it linear. This way, the model is akin to a Recurrent Neural Network (RNN): passing inputs for timestamp 0 and timestamp 1 together is the same as passing inputs at timestamp 0, then inputs at timestamp 1 along with the state of timestamp 0 (see example below).

Reviewer: My understanding is the model is an RNN (not just akin to one). 

```suggestion
It suggests a tweak in the traditional Transformer attention to make it linear. This way, the model can be used as recurrent network: passing inputs for timestamp 0 and timestamp 1 together is the same as passing inputs at timestamp 0, then inputs at timestamp 1 along with the state of timestamp 0 (see example below).
```

## Rwkv attention and the recurrent formulas

In a traditional autoregressive Transformer, attention is written as

$$O = \hbox{softmax}(QK^{T} / \sqrt{d}) V$$

> where \\(Q\\), \\(K\\) and \\(V\\) are matrices of shape `seq_len x hidden_size` named the Query, Key, and Value (in reality, these are matrices with higher dimensionality to account for a batch dimension and an attention head dimension. In this instance, we are only interested in the last two dimensions, which is where the matrix product is taken, so for the sake of simplicity we only consider those two). The product \\(QK^{T}\\) then has shape `seq_len x seq_len` and we can take the maxtrix product with \\(V\\) to get the output \\(O\\) of the same shape as the others.  

Reviewer: * The previous paragraph is easier to understand. We can still cut a few words from that though :) 
* I realise `query` etc are names we're giving, but they're written with lowercases everywhere in the library, so let's keep to that pattern 
* There's a missing noun after `Value` -> I can't say `they are the blue` (blue what?)

```suggestion
where \\(Q\\), \\(K\\) and \\(V\\) are matrices of shape `seq_len x hidden_size` named query, key and value (they are actually bigger matrices with a batch dimension and an attention head dimension but we're only interested in the last two, where the matrix product is taken, so for the sake of simplicity we only consider these). The product \\(QK^{T}\\) then has shape `seq_len x seq_len` and we can take the maxtrix product with \\(V\\) to get the output \\(O\\) of the same shape as the others.
```

Replacing the softmax by its value gives:

$$O_{i} = \frac{\sum_{j=1}^{i} e^{Q_{i} K_{j}^{T} / \sqrt{d}} V_{j}}{\sum_{j=1}^{i} e^{Q_{i} K_{j}^{T} / \sqrt{d}}}$$

> Note that the entries in \\(QK^{T}\\) corresponding to \\(j > i\\) are masked (the sum stops at \\(j)\\) because the attention is not allowed to look at future tokens (only past ones).

Reviewer: I think we need to end the latex formatting on the inner bracket here

```suggestion
Note that the entries in \\(QK^{T}\\) corresponding to \\(j > i\\) are masked (the sum stops at \\(j\\)) because the attention is not allowed to look at future tokens (only past ones).
```
